activestorage/app/models/active_storage/blob.rb

```diff
@@ -201,7 +201,7 @@ def representable?
   # with users. Instead, the +service_url+ should only be exposed as a redirect from a stable, possibly authenticated URL.
   # Hiding the +service_url+ behind a redirect also gives you the power to change services without updating all URLs. And
   # it allows permanent URLs that redirect to the +service_url+ to be cached in the view.
-  def service_url(expires_in: service.url_expires_in, disposition: :inline)
+  def service_url(expires_in: service.url_expires_in, disposition: :inline, filename: self.filename)
```

Comment:
This is the simplest version I came up with, that assumes a `ActiveStorage::Filename` instance here. I was thinking that a string could also be allowed, and then do: 
```
service.url ... filename: ActiveStorage::Filename.new(filename.to_s)
```
Or implement something like `ActiveStorage::Filename#wrap` and save doing the conversion twice in the default case:
```
service.url ... filename: ActiveStorage::Filename.wrap(filename)
```